lib/influxdb2/client/default_api.rb

```diff
@@ -32,7 +34,17 @@ def initialize(options:)
 
     private
 
-    def _post(payload, uri, limit = @max_redirect_count)
+    def _post_json(payload, uri, headers: {})
+      _check_arg_type(:headers, headers, Hash)
+      _post(payload, uri, headers: headers.merge({HEADER_CONTENT_TYPE => 'application/json'}))
+    end
+
+    def _post_text(payload, uri, headers: {})
+      _check_arg_type(:headers, headers, Hash)
+      _post(payload, uri, headers: headers.merge({HEADER_CONTENT_TYPE => 'text/plain'}))
+    end
+
+    def _post(payload, uri, limit: @max_redirect_count, headers: {})
```

Comment:
what was your rationale behind creating two separate methods, `_post_json` and `_post_text` which are identical aside from content-type? keeping the original `_post` method and exposing a `content_type` named argument w/ a reasonable default (e.g. `application/json`) might also be worth considering? (would also provide increased flexibility in allowing `_post` consumers to provide different content-type headers alongside their payloads if that's ever deemed beneficial)

(i guess, on second thought, not introducing another argument dependency in favor of encapsulating the content-type-setting logic in this class is probably more robust but still curious to hear your thoughts)

Reply:
The underlying `_post` method is HTTP-centric so it warranted a higher-level `headers` concept if we are to be introducing any additional headers. However, this introduced a minor inconvenience at the JSON-based call sites so I just added some helpers.